src/databricks/sql/result_set.py

```
@@ -480,46 +492,201 @@ def __init__(
         )
 
-    def _fill_results_buffer(self):
-        """Fill the results buffer from the backend."""
-        raise NotImplementedError(
-            "_fill_results_buffer is not implemented for SEA backend"
-        )
+    def _convert_json_to_arrow(self, rows: List) -> "pyarrow.Table":
+        """
+        Convert raw data rows to Arrow table.
+        """
+        if not rows:
+            return pyarrow.Table.from_pydict({})
+
+        columns = []
+        num_cols = len(rows[0])
+        for i in range(num_cols):
+            columns.append([row[i] for row in rows])
+        names = [col[0] for col in self.description]
+        return pyarrow.Table.from_arrays(columns, names=names)
+
+    def _convert_json_types(self, rows: List) -> List:
+        """
+        Convert raw data rows to Row objects with named columns based on description.
+        Also converts string values to appropriate Python types based on column metadata.
+        """
+
+        if not self.description or not rows:
+            return rows
+
+        # JSON + INLINE gives us string values, so we convert them to appropriate
+        #   types based on column metadata
+        converted_rows = []
+        for row in rows:
+            converted_row = []
+
+            for i, value in enumerate(row):
+                column_type = self.description[i][1]
+                precision = self.description[i][4]
+                scale = self.description[i][5]
+
+                try:
+                    converted_value = SqlTypeConverter.convert_value(
+                        value, column_type, precision=precision, scale=scale
+                    )
+                    converted_row.append(converted_value)
+                except Exception as e:
+                    logger.warning(
+                        f"Error converting value '{value}' to {column_type}: {e}"
+                    )
+                    converted_row.append(value)
```

we seem to be swallowing the exception and doing nothing. is it okay? will it have any unintended side-effects?